email-service/src/main/java/no/sikt/nva/email/EmailRequestHandler.java

```diff
@@ -35,14 +46,68 @@ public EmailRequestHandler() {
     }
 
     @Override
-    protected String processInput(EmailRequest input, RequestInfo requestInfo, Context context)
+    protected String processInput(EmailRequest emailRequest, RequestInfo requestInfo, Context context)
         throws ApiGatewayException {
-        logger.info("processing request");
-        return "hello world";
+        validateRequestBody(emailRequest);
+        var sendEmailResult = sendEmail(emailRequest);
+        logger.info(String.format(EMAIL_LOG_INFO_TRACK_ID, sendEmailResult.getMessageId()));
+        return SUCCESS_MESSAGE;
     }
 
     @Override
     protected Integer getSuccessStatusCode(EmailRequest input, String output) {
         return HttpURLConnection.HTTP_OK;
     }
+
+    private SendEmailResult sendEmail(EmailRequest emailRequest) throws EmailException {
+        return attempt(() -> createSesEmailRequest(emailRequest))
+                   .map(amazonSimpleEmailService::sendEmail)
+                   .orElseThrow(this::logFailureAndThrowEmailException);
+    }
+
+    @SuppressWarnings("PMD.InvalidLogMessageFormat")
+    private EmailException logFailureAndThrowEmailException(Failure<SendEmailResult> failure) {
+        logger.warn(COULD_NOT_SEND_EMAIL_MESSAGE, failure.getException());
+        return new EmailException(COULD_NOT_SEND_EMAIL_MESSAGE, HttpURLConnection.HTTP_INTERNAL_ERROR);
+    }
+
+    private SendEmailRequest createSesEmailRequest(EmailRequest emailRequest) {
+        return new SendEmailRequest()
+                   .withDestination(getDestination(emailRequest))
+                   .withMessage(getMessage(emailRequest))
+                   .withSource(emailRequest.getFromName())
+                   .withConfigurationSetName(CONFIGSET);
+    }
+
+    private Message getMessage(EmailRequest emailRequest) {
+        return new Message()
+                   .withBody(new Body()
+                                 .withHtml(createContent(emailRequest.getTextHtml()))
+                                 .withText(createContent(emailRequest.getText())))
+                   .withSubject(createContent(emailRequest.getSubject()));
+    }
+
+    private Content createContent(String data) {
+        return new Content()
+                   .withCharset(UTF_8)
+                   .withData(
+                       data);
+    }
+
+    private Destination getDestination(EmailRequest emailRequest) {
+        return new Destination()
+                   .withToAddresses(emailRequest.getToName())
+                   .withCcAddresses(emailRequest.getCc())
+                   .withBccAddresses(emailRequest.getBcc());
+    }
+
+    @SuppressWarnings("PMD.InvalidLogMessageFormat")
```

Comment:
why are you suppressing the warning

Reply:
codacy was complaining. There's no problem with the method locally. It's complaining that I have too many arguments. "Expected 0 arguments got 1"